app/Http/Middleware/CheckVideoOwnership.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use App\Models\Video;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use App\Http\Responses\BaseResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+class CheckVideoOwnership
+{
+/**
+ * Handle an incoming request.
+ *
+ * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+ */
+   public function handle(Request $request, Closure $next): Response
```

Comment:
Good approach for handling and passing video attribute to controller, minimizing the request call.